catalog/catalog.yaml

```diff
@@ -222,10 +222,33 @@ components:
           description: The prefix to be added to the API key value when it is sent in the header.
           x-go-type-skip-optional-pointer: true
 
+    ApiKeyAsBasicOpts:
+      type: object
+      description: when this object is present, it means that this provider uses Basic Auth to actually collect an API key
+      properties:
+        fieldUsed:
+          type: string
+          enum: [username, password]
+          example: username
+          description: whether the API key should be used as the username or password.
+          x-go-type-skip-optional-pointer: true
+        keyFormat:
+          type: string
+          example: "api:%s"
+          description: How to transform the API key in to a basic auth user:pass string. The %s is replaced with the API key value.
+          x-go-type-skip-optional-pointer: true
+
     BasicAuthOpts:
       type: object
       description: Configuration for Basic Auth. Optional.
       properties:
+        apiKeyLike:
```

Comment:
`apiKeyLike` can be removed now, since one can just check for the existence of `apiKeyAsBasicOpts`